### ISSUE TITLE
Missing Dependency Under Ubuntu

### ISSUE THREAD

**Issue by dsoprea** [walkr/oi#2]
The package complains about there being no _libhttpmsg_ library available and it doesn't seem to be [readily] available under Ubuntu:

```
Traceback (most recent call last):
  File "programd.py", line 2, in <module>
    import oi
  File "/usr/local/lib/python2.7/dist-packages/oi/__init__.py", line 1, in <module>
    from .core import *
  File "/usr/local/lib/python2.7/dist-packages/oi/core.py", line 8, in <module>
    from nanoservice import Service
  File "/usr/local/lib/python2.7/dist-packages/nanoservice/__init__.py", line 1, in <module>
    from .client import *
  File "/usr/local/lib/python2.7/dist-packages/nanoservice/client.py", line 27, in <module>
    import nanomsg
  File "/usr/local/lib/python2.7/dist-packages/nanomsg/__init__.py", line 7, in <module>
    from . import wrapper
  File "/usr/local/lib/python2.7/dist-packages/nanomsg/wrapper.py", line 4, in <module>
    _wrapper = _load_wrapper()
  File "/usr/local/lib/python2.7/dist-packages/nanomsg_wrappers/__init__.py", line 23, in load_wrapper
    return importlib.import_module('_nanomsg_ctypes')
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/usr/local/lib/python2.7/dist-packages/_nanomsg_ctypes/__init__.py", line 13, in <module>
    _lib = ctypes.cdll.LoadLibrary('libnanomsg.so')
  File "/usr/lib/python2.7/ctypes/__init__.py", line 443, in LoadLibrary
    return self._dlltype(name)
  File "/usr/lib/python2.7/ctypes/__init__.py", line 365, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: libnanomsg.so: cannot open shared object file: No such file or directory
```

**Comment by walkr** [comment]
To install [nanomsg](https://github.com/nanomsg/nanomsg) on your system try this:

``` shell
$ git clone git@github.com:nanomsg/nanomsg.git
$ cd nanomsg
$ ./configure
$ make
$ make check
$ sudo make install
```

**Comment by dsoprea** [comment]
When you clone directly from Github, you need to run "autogen.sh" before calling _configure._ However, this always failed. However, when you download one of the release-archives this is already done for you.

It looks like the package is only available as of 14.10 and I was only using 14.04 (in a Vagrant instance). I seemed to have trouble building, probably because I was on 14.04 . When I switched to using 15.04, I had the package and building wasn't necessary.

In order to install the _nanomsg_ Python package, I needed to install `python-dev` and `libnanomsg-dev`, obviously, to support the _cffi_ build.

What a pain, though I hadn't previously heard of _nanomsg_. I didn't realize that it existed nor did I know that it is supposed to supplant _0MQ_.

**Comment by walkr** [comment]
Yeah, I know, it sucks when one needs to go through all these hoops to install something.
